packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1912,7 +1912,9 @@ export class MergeTree {
 
 		const annotateSegment = (segment: ISegment) => {
 			assert(
-				!(reservedMarkerIdKey in props) || props.markerId === segment.properties?.markerId,
+				!(segment.type === "Marker") ||
```

Comment:
Use Marker.is. That is authoritative rather than duplicating that implementation here. Then you can use segment.getId() below instead of segment.properties?.markerId.